staff/joan-luis/bingo/index.js

```diff
@@ -0,0 +1,362 @@
+//Bingo Game.
+
+//const MAX_NUM_RND=99;
+//Declaration a object bingo card basic
+var oemBingoCardBasic = [
+    { number: 55, matched: false },
+    { number: 11, matched: false },
+    { number: 3, matched: false },
+    { number: 73, matched: false },
+    { number: 32, matched: false }  
+];
+
+var nuevoArray = new Array(5);
+nuevoArray[0] = new Array(5);
+nuevoArray[1] = new Array(5);
+nuevoArray[2]= new Array(5);
+
+// drclaration array with users
+var user=[{name:"", points: 0,}];
+var isMached=0;
+var numBalls=0; // permet porta un control de la quantitat de boles per tal de fer el calcul de punts
+var version;
+var machedLine=false;
+var machedBingo=false;
+var guessLineyet=0; 
+
+//Permite clonar un objeto
+function clone( obj ) {
+    if ( obj === null || typeof obj  !== 'object' ) {
+        return obj;
+    }
+ 
+    var temp = obj.constructor();
+    for ( var key in obj ) {
+        temp[ key ] = clone( obj[ key ] );
+    }
+ 
+    return temp;
+}
+
+function cardBingo(arrBingo){
+    var z=0;
+    auxRnd=getNumRandom(15);
+    for(i=0;i<3;i++){
+        for (j=0; j<5;j++){
+            arrBingo[i][j]={number: auxRnd[z], mached: false}; 
+            z++;
+        }
+    }
+    return arrBingo;   
+}
+
+
+//Function to generate a number random
+function getNumRandom(lenghtOfArr){
+    var numRndArr=[];
+    while(numRndArr.length<lenghtOfArr){
+        var auxNum=Math.floor((Math.random()*99)+1);
+        if (numRndArr.indexOf(auxNum)==-1){
+            numRndArr.push(auxNum);
+        };       
+    };
+    return numRndArr;
+};
+
+
+
+function printCard(objCard,version){
+    
+        if(version== 'b'){
+                console.log(objCard[0].number + " "+ objCard[1].number + " "+ objCard[2].number + " "+ objCard[3].number + " "+ objCard[4].number);
+        }else if(version=='p'){
+                console.log(objCard[0][0].number + " "+ objCard[0][1].number + " "+ objCard[0][2].number + " "+ objCard[0][3].number + " "+ objCard[0][4].number);
+                console.log(objCard[1][0].number + " "+ objCard[1][1].number + " "+ objCard[1][2].number + " "+ objCard[1][3].number + " "+ objCard[1][4].number);
+                console.log(objCard[2][0].number + " "+ objCard[2][1].number + " "+ objCard[2][2].number + " "+ objCard[2][3].number + " "+ objCard[2][4].number);
+        }
```

Comment:
En vez de hacer una tabla con cada cifra del bingo, haz una función para que te generes números random distintos para cada línea y columna, de este estilo:
>function getCard(rows, squares) { // el cartón del bingo
	rows = 3; // el cartón se rellenará con el número de filas que se quiera
	squares = 5; // cada fila se rellenará con el número de cifras que se quiera
	let arrayCard = [];
	for(let i = 0; i < rows; i++) {
    	arrayCard[i] = [];
		for(let j = 0; j < squares; j++) {
			arrayCard[i][j] = getNewNumber(cardNumsTaken);
		}
	}
return arrayCard;
}
let card = getCard();

De este modo, tu siempre tienes un número random, que puedes repartir en cada "celda" del cartón del bingo. En el ejemplo que pongo, con sólo cambiar filas y columnas no necesito añadir los números random uno por uno.